Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type or provide a default value, which allows to infer the type");
```

Comment:
(I honestly still find it a bit surprising that users can use `NSNumber` in Swift if a default value is provided. Is this intentional, or just a side effect of how our model parsing works? As a user, I wouldn't expect a default value of `1` would limit me to integers and `1.0` to floats.)